src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -223,44 +221,51 @@ public void ResetCheckpoint()
         /// </summary>
         public void Start()
         {
+            if (_disposed) {
+                throw new ObjectDisposedException(CouchbaseLiteErrorMessage.ReplicatorDisposed);
+            }
+
+            var status = default(C4ReplicatorStatus);
             DispatchQueue.DispatchSync(() =>
             {
-                if (_disposed) {
-                    throw new ObjectDisposedException(CouchbaseLiteErrorMessage.ReplicatorDisposed);
+                var err = SetupC4Replicator();
+                if (err.code > 0) {
+                    WriteLog.To.Sync.E(Tag, $"Setup replicator {this} failed.");
                 }
 
                 if (_repl != null) {
-                    WriteLog.To.Sync.W(Tag, $"{this} has already started");
-                    return;
+                    WriteLog.To.Sync.I(Tag, $"{this}: Starting");
+                    Native.c4repl_start(_repl);
+                    Config.Database.AddActiveReplication(this);
+                    status = Native.c4repl_getStatus(_repl);
+                } else {
+                    status = new C4ReplicatorStatus {
+                        error = err,
+                        level = C4ReplicatorActivityLevel.Stopped,
+                        progress = new C4Progress()
+                    };
                 }
-
-                WriteLog.To.Sync.I(Tag, $"{this}: Starting");
-                _retryCount = 0;
-                StartInternal();
             });
+
+            UpdateStateProperties(status);
+            DispatchQueue.DispatchSync(() => StatusChangedCallback(status));
         }
 
+
         /// <summary>
         /// Stops the replication
         /// </summary>
         public void Stop()
         {
             DispatchQueue.DispatchSync(() =>
             {
-                if (_stopping) {
-                    return;
-                }
-                
-                _stopping = true;
-                _reachability?.Stop();
```

Comment:
Why was this removed?  The reason it is here is so that the reachability callback does not interfere with the stopping process.

Reply:
StopReachabilityObserver() is called when replicator is in idle or busy and stopped states. Is `c4repl_setHostReachable` taking care Reachability Observing on the LiteCore side?

Reply:
All that method does is react to the information that is provided.  The platform still needs to monitor the actual condition and inform LiteCore.  If stop is called though, the monitoring should immediately stop.